streams/src/main/java/org/apache/kafka/streams/state/Stores.java

```diff
@@ -44,66 +49,120 @@ public static StoreFactory create(final String name) {
                 return new ValueFactory<K>() {
                     @Override
                     public <V> KeyValueFactory<K, V> withValues(final Serde<V> valueSerde) {
+
                         return new KeyValueFactory<K, V>() {
+                            public LoggedKeyValueFactory<K, V> logged(final String cleanupPolicy, final Map<String, String> config) {
+                                if (!InternalTopicManager.isValidCleanupPolicy(cleanupPolicy)) {
+                                    throw new IllegalArgumentException("cleanup policy: " + cleanupPolicy + " is not valid");
+                                }
+                                final Map<String, String> logConfig = new HashMap<>(config);
+                                logConfig.put("cleanup.policy", cleanupPolicy.toLowerCase(Locale.ROOT));
+                                return loggedKeyValueFactory(name, true, logConfig, keySerde, valueSerde);
+                            }
+
+                            public LoggedKeyValueFactory<K, V> notLogged() {
```

Comment:
I did this to avoid duplication. `enableCaching` is subtly different in that it only applies to `PersistentKeyValueFactory` whereas these calls apply to both `InMemory..` and `Persistent...`